arangod/Agency/Agent.cpp

```diff
@@ -2283,10 +2293,10 @@ query_t Agent::buildDB(arangodb::consensus::index_t index) {
 
   {
     READ_LOCKER(oLocker, _outputLock);
-    if (index > _commitIndex) {
+    if (index > _commitIndex.load(std::memory_order_relaxed)) {
       LOG_TOPIC("88754", INFO, Logger::AGENCY)
-          << "Cannot snapshot beyond leaderCommitIndex: " << _commitIndex;
-      index = _commitIndex;
+          << "Cannot snapshot beyond leaderCommitIndex: " << _commitIndex.load(std::memory_order_relaxed);
+      index = _commitIndex.load(std::memory_order_relaxed);
```

Comment:
would be more efficient to load the value of `_commitIndex` just once into a variable, and then use it from there.

Reply:
agree